core/core_modules/update.js

```diff
@@ -1,128 +1,148 @@
+                }
+                else {
+                    git.submoduleUpdate(function(err) {
+                        if (err, consoleOutput) {
+                            console.critical(err);
+                            console.debug('Periodic auto git submodule update failed. Manual intervention is probably required.\n Error: ' + consoleOutput);
+                        }
+                        else {
+                            try {
+                                install.update();
+                            }
+                            catch (e) {
+                                console.critical(e);
+                                api.sendMessage('Periodic auto update of NPM packages failed. Manual NPM intervention will be required.', event.thread_id);
+                            }
+                        }
 
-				if (checkForHash()) {
-					shutdown(StatusFlag.ShutdownShouldRestart);
-				}
-
-				setUpdateTimer();
-			});
-		}, updatePeriod);
-	};
+                        if (checkForHash()) {
+                            shutdown(StatusFlag.ShutdownShouldRestart);
+                        }
 
+                        setUpdateTimer();
+                    });
+                }
+            });
+        }, updatePeriod);
+    };
+	
 exports.match = function(event, commandPrefix) {
-	return event.body === commandPrefix + 'update';
+    return event.body === commandPrefix + 'update';
 };
 
 exports.load = function() {
-	var isEnabled = this.config.getConfig('update').autoUpdateEnabled,
+    var isEnabled = null,
         branchName;
 
+    isEnabled = this.config.getConfig('update').autoUpdateEnabled;
     shutdown = this.shutdown;
 
-	git.getCurrentBranchName(function (err, consoleOutput) {
-		if (err) {
-			console.debug('Failed to get current branch name\n Error: ' + consoleOutput);
-			return;
-		}
-		branchName = consoleOutput.trim();
-	});
+    git.getCurrentBranchName(function (err, consoleOutput) {
+        if (err) {
+            console.debug('Failed to get current branch name\n Error: ' + consoleOutput);
+            return;
+        }
+        branchName = consoleOutput.trim();
+    });
 
-	// Only allow auto update to be run on the master branch.
-	// By default auto update is enabled but can be turned off in the config.
-	if (branchName === 'master' && (isEnabled || isEnabled === undefined)) {
-		var configUpdatePeriod = this.config.getConfig('update').autoUpdatePeriod;
-		if (configUpdatePeriod) {
-			updatePeriod = configUpdatePeriod;
-		}
-		git.getSHAOfHead(function (err, consoleOutput) {
-			if (err) {
-				console.debug('Failed to get SHA hash of HEAD\n Error: ' + consoleOutput);
-				return;
-			}
-			currentSHA = consoleOutput;
-		});
-		setUpdateTimer();
-	}
+    // Only allow auto update to be run on the master branch.
+    // By default auto update is enabled but can be turned off in the config.
+    if (branchName === 'master' && (isEnabled || isEnabled === null)) {
```

Comment:
Redundant logic:
```
var a = null;
if (a) {
// never here
}
else {
console.log('we get here');
}
```

Simplify `(isEnabled || isEnabled === null)` to `isEnabled`